#!/usr/bin/env python
PACKAGE = "two_d_guidance"
    
from dynamic_reconfigure.parameter_generator_catkin import *
    
gen = ParameterGenerator()
    
gen.add("lookahead",    double_t, 0, "Look Ahead", .4,  0.01,   1.5)
gen.add("vel_sp",       double_t, 0, "Velocity setpoint", .4,  0.01,   2.5)

guidance_mode_enum = gen.enum([ gen.const("idle",   int_t, 0, "Idle"),
                                gen.const("drive",  int_t, 1, "Drive")],
                               "guidance mode")
   
gen.add("guidance_mode", int_t, 0, "guidance mode", 0, 0, 1, edit_method=guidance_mode_enum)
  
 
exit(gen.generate(PACKAGE, "two_d_guidance", "fl_guidance"))
